metashare/api/sf_org_changes.py

```diff
@@ -71,7 +126,7 @@ def run_retrieve_task(user, scratch_org, project_path, desired_changes):
     retrieve_components(
         components,
         org_config,
-        path,
+        target_directory,
         md_format,
         extra_package_xml_opts=package_xml_opts,
         namespace_tokenize=False,
```

Comment:
@davisagli I'm not entirely sure how to functionally test this, but in practice the `target_directory` arg doesn't seem to be making a difference here. I captured changes three times, each into different target directories, but every time the changes are in the `force-app/` directory:

- https://github.com/oddbird/MetaShare-Test/commit/b74d1bfc400b816cc1d11b86da646b1c6353ce6f
- https://github.com/oddbird/MetaShare-Test/commit/97e3cabee04909aa58114be6f00951cd5fc8419c
- https://github.com/oddbird/MetaShare-Test/commit/da4d575e33d197dbe25b39d4eb9e5d3490f41fd4

Maybe I'm looking for the wrong thing?

Reply:
@jgerigmeyer I looked at this a bit and see at least two problems:
1. sf_org_changes.run_retrieve_task is using the wrong logic for determining whether the directory is md_format or not (see this comment: https://github.com/oddbird/MetaShare/pull/191/files/332d480427985d10cbdedf5eaddc85407d10b458#r387230332) -- that leads to incorrectly determining that we're retrieving in sfdx format, which leads to...
2. I'm not sure if it's possible to retrieve components into a non-default sfdx format directory. I need to research if there's something we can pass to the underlying command to make this work; for now the target directory is just getting lost in this scenario and everything ends up in the default package path.

Btw, the .sfdx folder should not be getting captured. It's in .gitignore on MetaShare-Test master but maybe not in your project branch.

Reply:
@davisagli 

1. I think this is fixed -- take a look if you like.
2. Hm, I'm not 100% sure what "a non-default sfdx format directory" is. Is this something you want us to prevent for now on the MS side of things? Or wait on that until you have a chance to do further research/testing?
3. Actually, it looks like `.sfdx/` is in fact git-ignored on that branch (https://github.com/oddbird/MetaShare-Test/blob/feature/fresh-projects-get-your-projects-here__this-gets-cleared-yes/.gitignore#L2), but it's getting committed anyway. Is that a bug somewhere?

Reply:
2. It's a directory that is listed in packageDirectories in sfdx-project.json, but not the default package directory. I don't think we are doing this in our projects so it's an edge case we can come back to later.
3. Sounds like a bug. We should not be committing things that are in .gitignore

Reply:
@davisagli I tried to test this again to look into (3), but even after avoiding/fixing the edge cases being discussed in https://github.com/oddbird/MetaShare/commit/e6ce18131d0c16729a79f883e120d450402e9543#r37616332 and https://github.com/oddbird/MetaShare/commit/e6ce18131d0c16729a79f883e120d450402e9543#r37616357, I'm getting an error when trying to commit to an unpackaged directory:

```
id=no-job-id at=INFO time="2020-03-03 20:47:51.687012" tag=external module=worker msg="default: metashare.api.jobs.commit_changes_from_org(commit_message='Trying: \"unpackaged/pre/opportunity_record_types\"', desired_changes={'CustomField': ['Contact.Favorite_Color__c']}, scratch_org=<ScratchOrg: ScratchOrg object (M137Z1O)>, target_directory='unpackaged/pre/opportunity_record_types', user=<SimpleLazyObject: <User: jgerigmeyer>>) (8435b0f1-7139-4dee-82ef-bf5b5dac3d08)"
id=4b5753793af84117aa76847d5052368f at=INFO time="2020-03-03 20:47:51.693832" tag=external module=logging_middleware method=POST path=/api/scratch-orgs/M137Z1O/commit/ status=202 source_ip=127.0.0.1 user_agent='Mozilla/5.0 (Macintosh;=True Intel=True Mac=True OS=True X=True 10.15;=True rv:73.0)=True Gecko/20100101=True Firefox/73.0'=True time=0.05345654487609863 forwardedfor=None user=n10YoRw
HTTP POST /api/scratch-orgs/M137Z1O/commit/ 202 [0.06, 127.0.0.1:37088]
Traceback (most recent call last):
  File "/app/metashare/api/jobs.py", line 293, in commit_changes_from_org
    commit_changes_to_github(
  File "/app/metashare/api/sf_org_changes.py", line 140, in commit_changes_to_github
    run_retrieve_task(
  File "/app/metashare/api/sf_org_changes.py", line 115, in run_retrieve_task
    retrieve_components(
  File "/usr/local/lib/python3.8/site-packages/cumulusci/tasks/salesforce/sourcetracking.py", line 235, in retrieve_components
    sfdx(
  File "/usr/local/lib/python3.8/site-packages/cumulusci/core/sfdx.py", line 49, in sfdx
    raise Exception(f"Command exited with return code {p.returncode}")
Exception: Command exited with return code 1

id=no-job-id at=ERROR time="2020-03-03 20:48:07.105966" tag=external module=worker msg="Exception: Command exited with return code 1
Traceback (most recent call last):
  File \"/usr/local/lib/python3.8/site-packages/rq/worker.py\", line 884, in perform_job
    rv = job.perform()
  File \"/app/metashare/rq_worker.py\", line 12, in wrapper
    return orig(self)
  File \"/usr/local/lib/python3.8/site-packages/rq/job.py\", line 664, in perform
    self._result = self._execute()
  File \"/usr/local/lib/python3.8/site-packages/rq/job.py\", line 670, in _execute
    return self.func(*self.args, **self.kwargs)
  File \"/app/metashare/api/jobs.py\", line 293, in commit_changes_from_org
    commit_changes_to_github(
  File \"/app/metashare/api/sf_org_changes.py\", line 140, in commit_changes_to_github
    run_retrieve_task(
  File \"/app/metashare/api/sf_org_changes.py\", line 115, in run_retrieve_task
    retrieve_components(
  File \"/usr/local/lib/python3.8/site-packages/cumulusci/tasks/salesforce/sourcetracking.py\", line 235, in retrieve_components
    sfdx(
  File \"/usr/local/lib/python3.8/site-packages/cumulusci/core/sfdx.py\", line 49, in sfdx
    raise Exception(f\"Command exited with return code {p.returncode}\")
Exception: Command exited with return code 1
"
```

Reply:
@jgerigmeyer I'll dig into this tomorrow.

Reply:
@davisagli To clarify, I hit that error trying to capture into `unpackaged/pre/*` or `unpackaged/post/*` -- it worked fine capturing into `force-app` (https://github.com/oddbird/MetaShare-Test/commit/d2d8bb2869c51986e8a94e3088e8b6dbed1e5b59).

Reply:
Will be fixed by https://github.com/SFDO-Tooling/CumulusCI/pull/1591 once that makes it into MetaShare.